static/extensions/Gen1x/chess-ext.js

```diff
@@ -1,906 +1,1077 @@
+                    { opcode: 'getValidMoves', blockType: 'reporter', text: 'get all valid moves' },
+                    { opcode: 'getPieceAt', blockType: 'reporter', text: 'get piece at [SQUARE]', arguments: { SQUARE: { type: 'string', defaultValue: 'e4' } } },
+                    { opcode: 'getTurn', blockType: 'reporter', text: 'whose turn? (w/b)' },
+                    { opcode: 'getPGN', blockType: 'reporter', text: 'get game PGN' },
+                    { blockType: 'label', text: 'Advanced Stats' },
+                    { opcode: 'getPieceCount', blockType: 'reporter', text: 'count of [PIECE_TYPE] pieces for [COLOR]', arguments: { PIECE_TYPE: { type: 'string', menu: 'pieceTypeMenu', defaultValue: 'all' }, COLOR: { type: 'string', menu: 'colorMenu', defaultValue: 'w' } } },
+                    { opcode: 'getMaterialScore', blockType: 'reporter', text: 'material score for [COLOR]', arguments: { COLOR: { type: 'string', menu: 'colorMenu', defaultValue: 'w' } } },
+                    { opcode: 'getAttackersOfSquare', hideFromPalette: true, blockType: 'reporter', text: 'attackers of square [SQUARE] by [COLOR]', arguments: { SQUARE: { type: 'string', defaultValue: 'e4' }, COLOR: { type: 'string', menu: 'colorMenu', defaultValue: 'w' } } },
+                    { opcode: 'getPieceMoves', blockType: 'reporter', text: 'valid moves for piece at [SQUARE]', arguments: { SQUARE: { type: 'string', defaultValue: 'e2' } } },
+                    { opcode: 'getPositionScore', blockType: 'reporter', text: 'position evaluation score' },
+                    { opcode: 'getMoveHistory', blockType: 'reporter', text: 'get move history' },
+                    { opcode: 'getCapturedPieces', blockType: 'reporter', text: 'captured pieces by [COLOR]', arguments: { COLOR: { type: 'string', menu: 'colorMenu', defaultValue: 'w' } } }
+                ],
+                menus: {
+                    positionMenu: { acceptReporters: false, items: ['center', 'topleft', 'topright', 'bottomleft', 'bottomright'] },
+                    sizeMenu: { acceptReporters: false, items: ['small', 'medium', 'large', 'full'] },
+                    pieceTypeMenu: { acceptReporters: false, items: ['all', 'p', 'n', 'b', 'r', 'q', 'k'] },
+                    colorMenu: { acceptReporters: false, items: ['w', 'b'] },
+                    interactivityMenu: { items: ['true', 'false'] },
+                    tileTypeMenu: { items: ['light', 'dark'] },
+                    styleMenu: { items: ['emojis', 'icons'] },
+                    onOffMenu: { items: ['on', 'off'] },
+                    easingMenu: { items: ['linear', 'ease', 'ease-in', 'ease-out', 'ease-in-out'] },
+                    perspectiveMenu: { acceptReporters: true, items: ['white', 'black'] }
+                }
+            };
+        }
+        loadCpuEngine() {
+            if (this.cpuEngineLoaded) return;
+            fetch('https://rawcdn.githack.com/Gen1xLol/testassets/refs/heads/main/lozza.js')

```

Comment:
it would probably be best if this returned the fetch promise so that the vm awaits this, that way you can verify if the load succeeded or not just by checking `is CPU engine loaded?` afterwards, rather then needing a whole separate block to be added.